guardian/shortcuts.py

```diff
@@ -215,6 +223,14 @@ def get_users_with_perms(obj, attach_perms=False, with_superusers=False,
         else:
             user_filters = {'%s__content_object' % related_name: obj}
         qset = Q(**user_filters)
+        if only_with_perms is not None:
+            permission_qset = Q()
+            for permission in only_with_perms:
+                permission_qset |= Q(**{
+                    '%s__permission' % related_name: Permission.objects.get(
+                        content_type=ctype, codename=permission)
+                    })
+            qset &= permission_qset
```

Comment:
You should also restrict permissions on the `groups__%s__permission`. So I had in #246:

```
group_filters.update({
  'groups__%s__permission' % group_rel_name: permission,
})
```

That part seems missing here?